### DISCUSSION TITLE
Merging [@types/node] declarations with [@types/pouchdb] breaks on 6.4.1

### DISCUSSION THREAD

**Discussion by JasonAsmk** [DefinitelyTyped/DefinitelyTyped#68143] [Issues with a @types package]
### Description:
There seems to be an issue with typescript recognising [`buffer` in @types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/buffer.d.ts#L125) as the [`buffer` in @types/pouchdb](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/pouchdb-core/index.d.ts#L11). Last time this does not happen is with `@types/node: ^14.0.0` and `@types/pouch: 6.4.0`. The error looks like this and prevents building:

```
node_modules/@types/node/buffer.d.ts:98:14 - error TS2661: Cannot export 'Buffer'. Only local declarations can be exported from a module.

98     export { Buffer };
                ~~~~~~
```

### Suggested solution
Moving all global declarations in `@types/pouchdb-core/index.d.ts` inside the `PouchDB` namespace just below seems to solve the build error, but I don't think this is the correct solution.
Also removing the global `buffer` declaration from pouchdb-core also works, in my understanding, these types exist to fill in the case where `@types/node` are not available during compilation.

### Reproduction:
[types-pouchdb-issue](https://github.com/JasonAsmk/types-pouchdb-issue)

### Versions:
  - @types/pouchdb 6.4.2 (also 6.4.1)
  - @types/node ^18.0.0
  - typescript 5.2.X (also tested 4.x.x)

### Related Issues:
https://github.com/pouchdb/pouchdb/issues/8544

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
Add skipLibCheck to tsconfig?

**Reply by JasonAsmk** [reply]
Skipping lib check altogether works but it doesn't address the actual conflict.

**Reply by Semigradsky** [reply]
There are some discussions [here](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/54260) and [here](https://github.com/pouchdb/pouchdb/issues/8544) but no real solution.
